integration-docs: Update Github Actions for new doc format.

Niloth-p opened the pull request:
Fixes: Part of zulip#29592.

Separated out from #32354.


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

[Doc link](https://zulip.com/integrations/doc/github-actions)
![image](https://github.com/user-attachments/assets/e8521a4e-ad02-48eb-8cc8-0adfa11e4be8)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

Niloth-p commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,25 @@
+{start_tabs}
 
-See also the [GitHub integration](/integrations/doc/github).
+1. {!create-a-generic-bot.md!}
```
Removed the "Note down the bot's email address and API key." line that was appended to this instruction.

Niloth-p commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,25 @@
-See also the [GitHub integration](/integrations/doc/github).
+1. {!create-a-generic-bot.md!}
+
+1. To send Zulip notifications whenever your workflow runs, add the
```
This was previously in 2 separate instructions, that has now been clubbed together, as they don't make much sense separately.
We introduce the action that needs to be added, and the values with which it needs to be configured in the same instruction.

Niloth-p commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,25 @@
+
+### Related documentation
+
+* [Zulip GitHub Actions repository][repo]
```
Since the link to the repo was added into the second instruction (in the last update), this link has been edited to re-use it, by adding the 'repo' ref link.

Niloth-p commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,25 @@
+
+* [Zulip GitHub Actions repository][repo]
+
+* [Configuring the Zulip Send Message Action][README]
```
Configuring the Action -> Configuring the Zulip Send Message Action

Niloth-p reviewed (commented):
Marked the updates since last review.

timabbott commented:
I think this should really include steps that include the examples from https://github.com/zulip/github-actions-zulip/blob/main/send-message/README.md as the instructions for how to use this.

That's all you need to know, and the multiple steps of indirection that involve this not saying what to do and linking there are confusing.

(It's unlikely those instructions will change materially over the next 2 years, so it's OK to duplicate them if need be).

alya commented on templates/zerver/integrations/github-actions.md:
````diff
@@ -1,8 +1,54 @@
+         content: "I come not, friends, to steal away your hearts." # Markdown message{% endraw %}
+      ```
+
+1. Use the following template to send a direct message on Zulip.
````
```suggestion
1. Use the following template to send a direct message.
```

laurynmm commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,54 @@
 
-  * If you want to report the success or failure of an Action to Zulip.
-  * If you want to send output from your actions to Zulip.
+It's easy to send Zulip messages from GitHub Actions!
```
I might revise this to,  "Get notifications about GitHub Actions in Zulip!"

laurynmm commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,54 @@
+         api-key: ${{ secrets.ZULIP_API_KEY }}
+         email: "username@example.com" # email address corresponding to the API key
+         organization-url: "https://your-org.zulipchat.com"
+         to: "9" # user_id
```
Maybe `# zulip_user_id` for the comment here.

laurynmm commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,54 @@
+      with:
+         api-key: ${{ secrets.ZULIP_API_KEY }}
+         email: "username@example.com" # email address corresponding to the API key
+         organization-url: "https://your-org.zulipchat.com"
```
Yeah, I just wonder since this will be used in all documentation cases (Zulip Cloud org, self-hosted org, and the general `zulip.com` docs) that we want to use a more generic example URL.

Something like `"https://your-zulip-org.com"` maybe?

@alya - mentioning you to get your thoughts.

laurynmm commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,60 @@
+
+1.  To send Zulip notifications whenever your workflow runs, add the
+    `zulip/github-actions-zulip/send-message@v1` action to your GitHub
+    Actions workflow file, and set the input values as specified in the
```
I wonder about "as specified" here. The message content for the template is for a specific case. Might be better to be something more generic? Or note that the content would be for a specific case?

I assume that when setting these up, you would actually set these notifications to send whenever a particular thing happens when the workflow runs. The start of this instruction makes it sound like you're setting something to send every time the action runs.

laurynmm commented on templates/zerver/integrations/github-actions.md:
```diff
@@ -1,8 +1,60 @@
+      {% raw %}- name: Send a channel message
+      uses: zulip/github-actions-zulip/send-message@v1
+      with:
+         api-key: ${{ secrets.ZULIP_API_KEY }}
```
Does the user put the bot's API key inside the formatting here? Or is there a secrets file as part of the GitHub Actions set up? Or would the user put the API key string here in the template, like the email? If it's the latter, then I think it'd be clearer to just use a fake API key here.

laurynmm reviewed (commented):
@Niloth-p - Thanks for making those updates! I agree that the instructions for the two templates are clearer now.

I had a few more thoughts/questions on how we can make a few other things clearer. For the channel name, I think the screenshot and examples should match if possible.

Let me know if you have any questions about my feedback!

alya commented:
I pushed a commit that I think makes these instructions a lot more readable. But! Clicking on the DMs tab doesn't work. Can we fix that?

Note:
- I haven't tested whether comments with `#` are allowed, but it's Yaml, so they should be?

<img width="2118" height="1592" alt="Screenshot 2025-10-08 at 00 04 38@2x" src="https://github.com/user-attachments/assets/5659e417-f436-43d2-8f51-02795e418d5a" />

laurynmm commented:
I don't think we have the tab syntax set up for integration docs, so we'd have to implement that prior to those proposed changes.

Also, I think we've been trying to move away from comments in example code for integrations and API docs.

Niloth-p commented:
#### Updates

We've previously decided not to use code comments in the template. We actually had such a version 4 review rounds ago.
So, I've removed the code comments in @alya's edit.

I've absorbed other changes like:
- moving "workflow runs" to the intro line, and shaving off the first line of the second instruction.
- using tabbed sections.
- Adding an instruction to get the user's ID.
- Adding a link to GitHub workflows docs.

My newer changes:
- Added tabbed sections only for the example template.
- Added `<time:...>` syntax to match the screenshot.
- Added a link URL to the example to match the screenshot.
- Moved the tip outside the instruction blocks.

I've added the commit that adds tabbed section switching to integration docs - so that you can pull this branch and test it locally, even before it's merged.
I've opened a separate PR for that #36726, to split the discussion and the commits updating other docs to use tabbed sections.

<details>
<summary><h3>Screenshot</h3></summary>

<img width="901" height="1214" alt="image" src="https://github.com/user-attachments/assets/9b1b23ed-c42f-4afa-9b5b-743bae2a90b1" />
<img width="901" height="1214" alt="image" src="https://github.com/user-attachments/assets/3f38c9bc-0ddf-4d31-899b-0cace2336f57" />

</details>

Does this look reasonable? Or do we want to go back to earlier versions? @laurynmm @alya
